### PR TITLE
[Refactor] remove chain of view models

### PR DESCRIPTION
**Pull request by iaulakh**
## Purpose

* remove chain of view models

## Impact
*audio device selection, opening list of participant

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code
open audio device selection list from setup and call screen and confirms it works as before
open participant list from call screen and confirm it works as before

### CODE REVIEW

**Review by lpeixin — DISMISSED** [review]
tested

**Review by ShaunaSong — DISMISSED** [review]

File: azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/fragment/calling/controlbar/ControlBarView.kt [hunk 1]
```diff
@@ -141,6 +144,6 @@ internal class ControlBarView : LinearLayout {
     }
 
     private fun openAudioDeviceList() {
-        viewModel.displayAudioDeviceSelectionMenu()
+        openAudioDeviceSelectionMenuCallback()
```
Comment (phileo99) [inline azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/fragment/calling/controlbar/ControlBarView.kt:147]: Can we just get rid of openAudioDeviceList() and have the OnClickListener call openAudioDeviceSelectionMenuCallback() directly?

File: azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/fragment/calling/header/InfoHeaderView.kt [hunk 2]
```diff
@@ -64,6 +68,6 @@ internal class InfoHeaderView : ConstraintLayout {
     }
 
     private fun displayParticipantList() {
-        infoHeaderViewModel.displayParticipantList()
+        displayParticipantListCallback()
```
Comment (phileo99) [inline azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/fragment/calling/header/InfoHeaderView.kt:71]: Same thing here - this part of the code hasn't changed for a while, which means no new specs around 1displayParticipantsImageButton.setOnClickListener`, so can we remove `displayParticipantList()` and have the OnClickListener call `displayParticipantListCallback()` directly

File: azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/fragment/setup/components/SetupControlBarView.kt [hunk 3]
```diff
@@ -196,7 +199,7 @@ internal class SetupControlBarView : LinearLayout {
     }
 
     private fun openAudioDeviceList() {
-        viewModel.displayAudioDeviceSelectionMenu()
+        openAudioDeviceSelectionMenuCallback()
```
Comment (phileo99) [inline azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/fragment/setup/components/SetupControlBarView.kt:202]: Same thing, can we remove `openAudioDeviceList()` and  have the OnClickListener directly call `openAudioDeviceSelectionMenuCallback()`

**Review by phileo99 — CHANGES_REQUESTED** [review]

**Review by phileo99 — APPROVED** [review]
changes look good, tested on Espresso as well
👍

**Review by lpeixin — APPROVED** [review]

**Review by ShaunaSong — APPROVED** [review]

**Pull request merged**